### PR TITLE
Refactor SubmissionControl and robustify unit tests

### PR DESCRIPTION
**Pull request by clstrfsck**

### CODE REVIEW

File: radixdlt/src/main/java/com/radixdlt/mempool/MempoolReceiver.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import com.google.inject.Inject;
+import com.radixdlt.common.Atom;
+import com.radixdlt.network.MempoolNetworkRx;
+
+import io.reactivex.rxjava3.disposables.Disposable;
+
+/**
+ * Network glue for SubmissionControl.
+ */
+public final class MempoolReceiver {
```
Comment (talekhinezh) [inline radixdlt/src/main/java/com/radixdlt/mempool/MempoolReceiver.java:34]: Could this be the receiver from both mempool and the api?

Comment (talekhinezh) [reply radixdlt/src/main/java/com/radixdlt/mempool/MempoolReceiver.java:34]: And so all submissions can be contained in a single backpressured flow?

Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/mempool/MempoolReceiver.java:34]: Yeah, could be.  Let me have a look.

Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/mempool/MempoolReceiver.java:34]: Hmm.  Need to discuss.  Will ping you tomorrow.

**Pull request merged**